src/tox/session/commands/run/parallel.py

```diff
@@ -1,13 +1,23 @@
 import os
 import subprocess
 import sys
+import tempfile
```

Comment:
I believe this is already implemented partially within Action, we just need to make this an Action and perhaps have a flag allowing this👍

Reply:
this is way way simpler than involving `Action`

Reply:
I know it's simpler but not using the Action is a bug, as generally, that's what we **should** use. That was added especially for such use cases, and somewhere we even have an issue to make this content available as a file afterwards (for CIs); so that's the right solution to solve both at the same time.

Reply:
that seems orthogonal and existing :P

Reply:
not really orthogonal given in sense that the buffer overload is what is breaking here things, and would we write to file that would not exist 🤔

Reply:
we're talking small fix for a showstopping bug vs ~full rewrite of `--parallel` (and `Action` to some extent to make it threadsafe).  I'd much rather fix a bug than rewrite especially when the rewrite is likely to change and break behaviour.

Reply:
I don't believe making the change requires a full rewrite, probably just switching over to action and passing some flags along 🤔

Reply:
I started looking at it and it requires passing enough state into this thread to build an Action which is no small task

Reply:
isn't just tox_env.new_action ? https://github.com/tox-dev/tox/blob/master/src/tox/venv.py#L116 🤔

Reply:
it's not that simple, no -- `tox -p` is implemented by running sub process which invoke `python -m tox`: https://github.com/tox-dev/tox/pull/1202/files#diff-7f1e93c77a2f9f1b1f2c01a8950bad0dR25

Reply:
I know, I did it, bur action also uses underneath subprocess via popen.

Reply:
Anyways we can go ahead just mind you we'll need it switch at some point. Thought if not too hard we could do it now. If you find it non trivial we can hold for now.

Reply:
my initial try ended in a different deadlock due to the log files, I'd rather not

Reply:
let's come back to this later then 👍